src/models/wallet.js

```diff
@@ -0,0 +1,55 @@
+var StellarBase = require('stellar-base');
+var Ed25519 = require('ed25519');
+var Sjcl = require('sjcl');
+
+var DbUtil = require('../utils/db');
+var CryptoUtil = require('../utils/crypto');
+
+var wallet = {
+    
+    create: function(msisdn, pin) {
+        var salt = Sjcl.codec.base64.fromBits(Sjcl.random.randomWords(64/4));
+        var key = this._generateKeyPair();
+        var privateKey = key.privateKey;
+        var publicKey = key.publicKey;
+        var encryptionKey = pin;
+        var pinHash = CryptoUtil.hash(pin, salt);
+        var privateKeyEncrypted = CryptoUtil.encryptData(privateKey, encryptionKey);
+        var address = this.addressFromPublicKey(publicKey);
+
+        return DbUtil.promiseConnection()
+            .then(DbUtil.chainedQuery(
+                "INSERT INTO wallet ( msisdn, address, publickey, privatekey, pinhash, salt) VALUES ( $1, $2, $3, $4, $5, $6)",
```

Comment:
Would you consider using [Sequelize](http://docs.sequelizejs.com/en/latest/)?  Looks like a nicely built promises based ORM for Node.js.